addons/behaviour_toolkit/finite_state_machine/fsm_state_integrated_bt.gd

```diff
@@ -1,30 +1,68 @@
 @icon("res://addons/behaviour_toolkit/icons/FSMStateIntegration.svg")
 class_name FSMStateIntegratedBT extends FSMState
+## [FSMState] that can handle and activate [BTRoot] as it's child.
+##
+## When this [FSMStateIntegratedBT] state is entered it activates it's first
+## child of type [BTRoot].
+## [br][br]
+## If its [BTRoot] is swaped for another [BTRoot] at run-time, the new child
+## will be deactivated by default and, if not activated manually, will activate
+## automatically next time this state is entered.
 
 
+@export var fire_event_on_status: bool = false
+@export var on_status: BTBehaviour.BTStatus = BTBehaviour.BTStatus.SUCCESS
+@export var event: String
+
+
+# Connecting signal using @onready to omit the need to use super() call
+# in _ready() of extended nodes if they override _ready().
+@onready var __connect_disable_autostart: int = \
+	ready.connect(_disable_autostart)
+@onready var __connect_child_order_changed: int = \
+	child_order_changed.connect(_on_child_order_changed)
 
 @onready var behaviour_tree: BTRoot = _get_behaviour_tree()
 
 
-@export var fire_event_on_status: bool = false
-@export var on_status: BTBehaviour.BTStatus = BTBehaviour.BTStatus.SUCCESS
-@export var event: String
+## Swap this states current Behaviour Tree with the provided one.
+## If state has no [BTRoot] as a child the provided one will be added.
+## [br][br]
+## Old BTree is freed and the new BTree will be active next time the state will
+## be entered.
+func swap_behaviour_tree(btree: BTRoot,
+	force_readable_name: bool = false) -> void:
+	
+	if behaviour_tree == null:
+		add_child(btree, force_readable_name)
+	else:
+		behaviour_tree.queue_free()
+		add_child(btree, force_readable_name)
```

Comment:
Just realised you could do this. When adding nodes using the UI, I iterate over the nodes manually. May change this in the future.